.github/workflows/release.yml

```diff
@@ -0,0 +1,37 @@
+name: Release
+
+on:
+  push:
+    tags: [v*]
+
+jobs:
+  sdist:
+    runs-on: ubuntu-latest
+    environment: release
+
+    permissions:
+      contents: write # Used to authenticate github release publish
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.19'
+
+    - name: Generate sdist
+      run: |
+        VERSION=${GITHUB_REF_NAME#v}
+        go mod vendor
+        tar czf redis_exporter-${VERSION}.tar.gz --transform "s,^,redis_exporter-${VERSION}/," *
```

Comment:
Can you also add the equivalent for creating a zip file and attach it to the release? 
In the end this needs to be identical to the current process and that includes a tar.gz and a zip file in the release.

<img width="256" alt="image" src="https://github.com/user-attachments/assets/c60f618a-dab2-494d-bf77-226ab2afd4e3">

Reply:
done, and I've decided to also add `tar.xz`, which uses much better compression, for specific downstreams which wants smallest source tarball.